vktrace/vktrace_layer/vktrace_lib_trace.cpp

```diff
@@ -247,7 +254,23 @@ VKTRACER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL __HOOKED_vkMapMemory(VkDevice dev
         if (pInfo != NULL) {
             pInfo->ObjectInfo.DeviceMemory.mappedOffset = offset;
             pInfo->ObjectInfo.DeviceMemory.mappedSize = size;
-            pInfo->ObjectInfo.DeviceMemory.mappedAddress = *ppData;
+
+            // Page guard handling create a shadow memory for every mapped
+            // memory object and add page guard to capture the access of
+            // write and read, so the page guard handling can keep dual
+            // direction of sync between real mapped memory and the shadow
+            // memory.
+            //
+            // When starting trim, trim process need to read and save all image
+            // and buffer data to trace file, that will trigger all page guard
+            // on those memory if trim access shadow memory, and the status of
+            // page guard for those memory will be different with capture
+            // without trim. It causes corruption for some title if trim at
+            // some locations.
+            //
+            // So here we make trim to use real memory pointer to avoid change
+            // the pageguard status when PMB enabled.
+            pInfo->ObjectInfo.DeviceMemory.mappedAddress = pRealMappedData;
```

Comment:
Does mappedAddress get reset to the address of the shadow memory when we are done trimming and start tracing? It seems it would ned to be to properly track pmb changes after trimming is done. I couldn't find where that is done.

Reply:
The info in pInfo->ObjectInfo including DeviceMemory.mappedAddress is only used by trim module. the change is trim only, it doesn't need to reset after trim because PMB capture part has no change,  still use shadow memory pointer to track pmb change.